code/Model/EditableFormField/EditableFileField.php

```diff
@@ -60,14 +64,21 @@ public function getCMSFields()
             )
         );
 
-        $fields->addFieldToTab("Root.Main", LiteralField::create(
-            'FileUploadWarning',
-            '<p class="alert alert-info">' . _t(
-                'SilverStripe\\UserForms\\Model\\UserDefinedForm.FileUploadWarning',
-                'Files uploaded through this field could be publicly accessible if the exact URL is known'
-            )
-            . '</p>'
-        ), 'Type');
+        $folderIsRestricted = Member::actAs(null, function () {
```

Comment:
We've switched to only doing a permissions-only check

Previously we were using using `Member::actAs(null` which effectively checks if a file in in the public or protected asset store.  This will:
- check permissions
- check if the file has been published or not

Since there's two ways to move a file between stores, this was deemed as too confusing so we've switched to doing permission-only check for "$folderIsRestricted"

You'll want to get this the related assets PR merged and make use of that instead
https://github.com/silverstripe/silverstripe-assets/pull/398/files#diff-41b5ea2bbaadc935cf6b51194eba6e68R502

You'll also need to update this PR to either
- make silverstripe/assets:^1.6 a composer requirement, or
- (much easier) `if (method_exists($folder, 'hasRestrictedPermissions'))`

Reply:
I think it's cleaner to just bump the assets requirements to 1.6 rather than check if a method exists.

Reply:
Agree the ^1.6 is cleaner

However it's also quite annoying and not entirely practical since 1.6 assets hasn't yet been released.  It means, that, at this stage, a simple composer install will fail.  You need to alias "silverstripe/assets": "1.x-dev as 1.6", which is doable in your project, but it's no good for travis

Reply:
I've bumped the core requirement to `^4.6`. This shouldn't result in install failures, as all of our modules specify `"minimum-stability": "dev"` to allow installation of unstable packages.